cmds/core/dhclient/dhclient.go

```diff
@@ -47,57 +48,70 @@ var (
 	v4Port = flag.Int("v4-port", dhcpv4.ServerPort, "DHCPv4 server port to send to")
 )
 
-func main() {
-	flag.Parse()
-	if len(flag.Args()) > 1 {
-		log.Fatalf("only one re")
+func run(args []string, timeout int, retry int, dryRun bool, verbose bool, vverbose bool, ipv4 bool, ipv6 bool, v6Port int, v6Server string, v4Port int) error {
+	if len(args) > 1 {
+		return fmt.Errorf("more than one interface specified")
 	}
 
-	if len(flag.Args()) > 0 {
-		ifName = flag.Args()[0]
+	if len(args) > 0 {
+		ifName = args[0]
 	}
 
 	filteredIfs, err := dhclient.Interfaces(ifName)
 	if err != nil {
-		log.Fatal(err)
+		return err
+	}
+
+	if err := configureAll(filteredIfs, timeout, retry, dryRun, verbose, vverbose, ipv4, ipv6, v6Port, v6Server, v4Port); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func main() {
+	flag.Parse()
+	args := flag.Args()
+	if err := run(args, *timeout, *retry, *dryRun, *verbose, *vverbose, *ipv4, *ipv6, *v6Port, *v6Server, *v4Port); err != nil {
+		log.Fatalf("dhclient: %v", err)
 	}
 
-	configureAll(filteredIfs)
 }
 
-func configureAll(ifs []netlink.Link) {
-	packetTimeout := time.Duration(*timeout) * time.Second
+func configureAll(ifs []netlink.Link, timeout int, retry int, dryRun bool, verbose bool, vverbose bool, ipv4 bool, ipv6 bool, v6Port int, v6Server string, v4Port int) error {
+	packetTimeout := time.Duration(timeout) * time.Second
 
 	c := dhclient.Config{
 		Timeout: packetTimeout,
-		Retries: *retry,
+		Retries: retry,
 		V4ServerAddr: &net.UDPAddr{
 			IP:   net.IPv4bcast,
-			Port: *v4Port,
+			Port: v4Port,
 		},
 		V6ServerAddr: &net.UDPAddr{
-			IP:   net.ParseIP(*v6Server),
-			Port: *v6Port,
+			IP:   net.ParseIP(v6Server),
+			Port: v6Port,
 		},
 	}
-	if *verbose {
+	if verbose {
 		c.LogLevel = dhclient.LogSummary
 	}
-	if *vverbose {
+	if vverbose {
 		c.LogLevel = dhclient.LogDebug
 	}
-	r := dhclient.SendRequests(context.Background(), ifs, *ipv4, *ipv6, c, 30*time.Second)
+	r := dhclient.SendRequests(context.Background(), ifs, ipv4, ipv6, c, 30*time.Second)
 
 	for result := range r {
 		if result.Err != nil {
-			log.Printf("Could not configure %s for %s: %v", result.Interface.Attrs().Name, result.Protocol, result.Err)
-		} else if *dryRun {
+			return fmt.Errorf("Could not configure %s for %s: %w", result.Interface.Attrs().Name, result.Protocol, result.Err)
+		} else if dryRun {
 			log.Printf("Dry run: would have configured %s with %s", result.Interface.Attrs().Name, result.Lease)
 		} else if err := result.Lease.Configure(); err != nil {
-			log.Printf("Could not configure %s for %s: %v", result.Interface.Attrs().Name, result.Protocol, err)
+			return fmt.Errorf("Could not configure %s for %s: %w", result.Interface.Attrs().Name, result.Protocol, err)
```

Comment:
This is not a change in behavior I can support (both the log changes).

If one interface cannot be configured, a DHCP boot process (or dhclient) must be resilient and move on to the next interface, not fail and return.

Reply:
Thanks for the review. You are right, this was an oversight by me. I reverted the change in the latest commit.